### DISCUSSION TITLE
Confirm exit from page with unsaved changes doesn't show

### DISCUSSION THREAD

**Discussion by zoodogood** [community/community#86114] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Being able to make a quick change to a file is great. But sometimes it's so addictive that it takes you a long time to write out your commit. 
I lose unsaved changes by inadvertently triggering **Back** in the browser. It's an old problem. It's a reliability problem. 
It occurs on my phone too and is not tied to a specific browser.

Github has this warning if I try to close the page or click a link, but it doesn't guarantee all cases.

Please be understanding. If it is not just me that is having this problem, it should be resolved.

Perhaps the files being edited will be saved to local storage, every 15s. (continue editing)?

Steps to reproduce:
1) Go to edit the file in your browser, for example,
https://github.com/withastro/docs/edit/main/src/content/docs/en/core-concepts/rendering-modes.mdx;
2) Make a few changes in text;
3) Use your browser navigation to go back;
4) When you want to go back to edit the file, you will see that your changes have been canceled

Regards, thank you for your hard work

**Comment by Shahnawazk7944** [comment]
I understand your frustration with losing unsaved changes on GitHub, and you're definitely not alone. This is a common issue that many developers face, and it can be incredibly disruptive to workflow.

I appreciate your suggestions for potential solutions, and I think both temporary autosaving and a more comprehensive warning system have merit. Here's a breakdown of your ideas and some additional thoughts:

**Temporary Autosaving:**

* **Pros:**
    * Can prevent data loss from accidental actions like browser back button or navigation.
    * Provides peace of mind, knowing work is being saved incrementally.
* **Cons:**
    * May introduce performance overhead, especially on slower connections.
    * Could lead to confusion if changes haven't been reviewed before being saved.

**Comprehensive Warning System:**

* **Pros:**
    * Can alert users to potential data loss in more nuanced situations, like clicking specific links.
    * Offers users a chance to confirm their intentions before proceeding.
* **Cons:**
    * May become intrusive if triggered too frequently.
    * Requires careful design to avoid being annoying or disruptive to workflow.

**Additional Solutions:**

* **Draft functionality:** GitHub could implement a draft system for edits, allowing users to save unfinished work without committing it permanently.
* **Version control integration:** Tighter integration with version control systems could allow for easier recovery of lost changes.
* **Improved commit workflow:** Streamlining the commit process itself could encourage users to commit more frequently, reducing the risk of losing large amounts of work.

I hope this information is helpful! Remember, your feedback is valuable to GitHub, so don't hesitate to share your experiences and suggestions with them.
